File: Selnium script for testing/Script with Python/test_scenario_54.py
from selenium import webdriver
from selenium.webdriver.common.by import By
import time

def execute_test_54():
    """
    Dummy test run for scenario 54
    """
    driver = webdriver.Chrome()
    driver.implicitly_wait(10)
    driver.get("https://dummy.example.com/page_54")

    driver.find_element(By.ID, "user_name").send_keys("test_data_388")
    assert driver.find_element(By.ID, "success_msg").is_displayed()
    driver.find_element(By.ID, "checkout_btn").click()
    assert driver.find_element(By.ID, "success_msg").is_displayed()
    time.sleep(2)
    driver.find_element(By.CLASS_NAME, "nav-item").click()
    time.sleep(2)
    time.sleep(2)
    driver.find_element(By.LINK_TEXT, "Log Out").click()
    time.sleep(2)
    driver.find_element(By.PARTIAL_LINK_TEXT, "Profile").click()
    driver.find_element(By.XPATH, "//button[@type='submit']").click()

    driver.quit()

if __name__ == "__main__":
    execute_test_54()

File: Selnium script for testing/Script with Java/src/test/java/testscenario/TestScenario54.java
package testscenario;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import java.time.Duration;

public class TestScenario54 {
    public static void executeTest54() {
        /*
         * Dummy test run for scenario 54
         */
        WebDriver driver = new ChromeDriver();
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
        driver.get("https://dummy.example.com/page_54");

        try {
            if (!driver.getTitle().contains("Dashboard")) {
                System.out.println("Dashboard not in title");
            }
            driver.findElement(By.linkText("Log Out")).click();
            if (!driver.findElement(By.name("password_field")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
            driver.findElement(By.cssSelector(".promo-code")).click();
            driver.findElement(By.id("user_name")).sendKeys("test_data_556");
            driver.findElement(By.id("checkout_btn")).click();
            driver.findElement(By.className("nav-item")).click();
            driver.findElement(By.xpath("//button[@type='submit']")).click();
            if (!driver.findElement(By.id("user_name")).isDisplayed()) {
                System.out.println("Element not displayed");
            }
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
            driver.quit();
        }
    }
    
    public static void main(String[] args) {
        executeTest54();
    }
}
